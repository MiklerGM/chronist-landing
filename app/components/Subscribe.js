import React from 'react';

export default class Subscribe extends React.Component {
  constructor(props) {
    super(props);
    this.state = { email: '', error: '' };
  }

  render() {
    return (
    <div id="Subscribe" className="bg-2 text-center">
    <p>Получайте информацию о ходе проекта и датах запуска.</p>
    <form className="form-inline" action="email.php" onSubmit={(e) => {
      e.preventDefault();
      axios.post('/email.php', `email=${this.state.email}`)
      .then(function (response) {
        console.log(response);
      })
      .catch(function (error) {
        console.log(error);
      });
    }}>
      <div className="form-group">
        <div className="input-group">
          <input type="email" value={this.state.email} className="form-control" size="20" placeholder="Ваш E-mail" required onChange={(e) => {
            this.setState({...this.state, email: e.target.value });
          }} />
          <div className="input-group-btn">
            <button type="submit" className="btn btn-danger">Подписаться</button>
          </div>
        </div>
      </div>
      <div className="form-group">
        <a href="https://vk.com/public139815700" alt="Вконтакте"><button type="button" className="btn btn-primary"> <i className="fa fa-vk" aria-hidden="true"></i> Вконтакте </button></a>
      </div>
    </form>

    </div>
  );
  }
}
