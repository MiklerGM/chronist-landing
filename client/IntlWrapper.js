import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';

import localeDataRU from './locales/ru.json';
// import localeDataEN from './locales/en.json';

class IntlWrapper extends React.Component {
  render() {
    return (
      <IntlProvider
        locale={this.props.locale}
        key={this.props.locale}
        messages={this.props.messages}
      >
        {this.props.children}
      </IntlProvider>
    );
  }
}

IntlWrapper.propTypes = {
  locale: PropTypes.string,
  messages: PropTypes.object
};

IntlWrapper.defaultProps = {
  locale: 'ru',
  messages: localeDataRU
};

export default IntlWrapper;
