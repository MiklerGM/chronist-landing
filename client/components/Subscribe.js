import React from 'react';
import { FormattedMessage } from 'react-intl';
import axios from 'axios';

import { MailInput } from './Input';

const SocialButton = () => (
  <div className='social--button'>
    <a className='decorless' href='https://vk.com/chronist' alt='Вконтакте'>
      <button type='button' className='vk'>
        <i className='icon-vkontakte' aria-hidden='true' />
        <span><FormattedMessage id='home.subscribe.vk' /></span>
      </button>
    </a>
  </div>
);

class Subscribe extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      visibile: false,
      success: false,
      data: {
        email: '',
      }
    };
  }

  onSubmit(e) {
    e.preventDefault();
    const _this = this;
    axios.post('/shared/subscribe', `email=${this.state.data.email}`)
      .then((response) => {
        const success = response.status === 200;
        const wipe = success ? { email: '' } : {};
        _this.setState({ ..._this.state, ...wipe, visibile: true, success });
      })
      .catch((error) => {
        _this.setState({ ..._this.state, visibile: true, success: false });
        console.log(error);
      });
    return false;
  }

  getGlyph() {
    return this.state.success ? 'fa icon-check ' : 'fa icon-cancel';
  }

  handleInput(value) {
    this.setState({ data: value });
  }

  render() {
    return (
      <div className='page--segment text-center bg-gray'>
        <div className='page--content'>
          <p>
            <span>
              <FormattedMessage
                id='home.subscribe.message'
              />
            </span>
          </p>
          <div className='join-us'>
            <div>
              <form
                action='/shared/subscribe'
                method='POST'
                className='subscribe input-group'
                onSubmit={e => this.onSubmit(e)}
              >
                <MailInput
                  value={this.state.data.email}
                  name='email'
                  placeholder='contact.placeholder.email'
                  cb={e => this.handleInput(e)}
                />
                <button type='submit' className='red'>
                  <span><FormattedMessage id='home.subscribe.button' /></span>
                </button>
              </form>
            </div>
            <SocialButton />
          </div>
          <div
            key='result'
            style={this.state.visibile ? {} : { display: 'none' }}
            className={this.getGlyph()}
          >
            {' '}
            <span>
              {this.state.success
                ? <FormattedMessage id='home.subscribe.success' />
                : <FormattedMessage id='home.subscribe.fail' />}
            </span>
          </div>
        </div>
      </div>
    );
  }
}

export default Subscribe;
