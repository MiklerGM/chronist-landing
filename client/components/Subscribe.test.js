import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import IntlWrapper from '../IntlWrapper';
import Subscribe from './Subscribe';

const TestComponent = () => (
  <IntlWrapper>
    <Subscribe />
  </IntlWrapper>
);

// it must render
it('Subscribe render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

it('Subscribe render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

// Snapshot tests
it('Subscribe Snapshot', () => {
  const tree = renderer
    .create(<TestComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
