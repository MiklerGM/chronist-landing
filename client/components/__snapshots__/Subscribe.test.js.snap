// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscribe Snapshot 1`] = `
<div
  className="page--segment text-center bg-gray"
>
  <div
    className="page--content"
  >
    <p>
      <span>
        Чтобы быть в курсе наших обновлений - подпишитесь на рассылку или участвуйте в наших сообществах в соцсетях.
      </span>
    </p>
    <div
      className="join-us"
    >
      <div>
        <form
          action="/shared/subscribe"
          className="subscribe input-group"
          method="POST"
          onSubmit={[Function]}
        >
          <div
            className="input--cell"
          >
            <input
              className=""
              onChange={[Function]}
              onInvalid={[Function]}
              pattern="^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\\\.[a-z]{2,4}$"
              placeholder="Электронная почта"
              required={true}
              size="25"
              style={
                Object {
                  "gridRow": "1 / 2",
                }
              }
              type="email"
              value=""
            />
            <span
              className="invalid-message__hidden"
              style={
                Object {
                  "gridRow": "2 / 3",
                }
              }
            >
              <span>
                Неверный формат адреса электронной почты
              </span>
            </span>
          </div>
          <button
            className="red"
            type="submit"
          >
            <span>
              Подписаться
            </span>
          </button>
        </form>
      </div>
      <div
        className="social--button"
      >
        <a
          alt="Вконтакте"
          className="decorless"
          href="https://vk.com/chronist"
        >
          <button
            className="vk"
            type="button"
          >
            <i
              aria-hidden="true"
              className="icon-vkontakte"
            />
            <span>
              ВКонтакте
            </span>
          </button>
        </a>
      </div>
    </div>
    <div
      className="fa icon-cancel"
      style={
        Object {
          "display": "none",
        }
      }
    >
       
      <span>
        Произошла ошибка
      </span>
    </div>
  </div>
</div>
`;
